internal/http/http.go

```diff
@@ -28,3 +31,43 @@ func SetCacheHeaders(w http.ResponseWriter, maxAge time.Duration, timeUntilExpir
 	w.Header().Set("Cache-Control", "public, max-age="+maxAge.String())
 	w.Header().Set("Expires", time.Now().Add(timeUntilExpiry).Format(time.RFC1123))
 }
+
+func LimitRead(r io.ReadCloser, limit int64) ([]byte, error) {
```

Comment:
@hiddeco this was your trick, which has already been used elsewhere, moved here because it's generally useful. Do you mind looking?

Reply:
I think the placement in `http` is weird, and we should likely create an `io` package. Also: a doc block would be welcome.